### ISSUE TITLE
Query on segmentation modeling (deeplabv3)

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#874]
1. In general, shouldn't the `input size` and `target size` match? Why I need to downscale the target samples for training? For example,  input size 128, target size 32.
2. I got the following warning message. Tested on colab.  

> WARNING:tensorflow:Gradients do not exist for variables ['deep_lab_v3/feature_pyramid/output_P3/kernel:0', 'deep_lab_v3/feature_pyramid/output_P3/bias:0', 'deep_lab_v3/feature_pyramid/output_P4/kernel:0', 'deep_lab_v3/feature_pyramid/output_P4/bias:0', 'deep_lab_v3/feature_pyramid/output_P5/kernel:0', 'deep_lab_v3/feature_pyramid/output_P5/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?

3. `mixed precision` makes loss value to `nan` (probably need to cast the output of segmentation head to `tf.float32`. You might want to introduce `dtype` param in [here](https://github.com/keras-team/keras-cv/blob/673e63cf2fc5b4ff295004a20aac7b5e2544c16d/keras_cv/models/segmentation/__internal__/segmentation_head.py#L51).) 

[Reproducible Code.](https://colab.research.google.com/drive/1x-mB_CPnQkuqSMq7peHWk32h6V9Qi6h3?usp=sharing)

**Comment by innat** [comment]
cc. @qlzh727

**Comment by qlzh727** [comment]
Thanks for the question.

> 1. In general, shouldn't the `input size` and `target size` match? Why I need to downscale the target samples for training? For example,  input size 128, target size 32.

Sure. Currently the model is just return the raw output from detection head. We could add extra layers to scale up the output to let it match the input shape. (for better comparison).
> 2. I got the following warning message. Tested on colab.
> 
> > WARNING:tensorflow:Gradients do not exist for variables ['deep_lab_v3/feature_pyramid/output_P3/kernel:0', 'deep_lab_v3/feature_pyramid/output_P3/bias:0', 'deep_lab_v3/feature_pyramid/output_P4/kernel:0', 'deep_lab_v3/feature_pyramid/output_P4/bias:0', 'deep_lab_v3/feature_pyramid/output_P5/kernel:0', 'deep_lab_v3/feature_pyramid/output_P5/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?
> 

Let me take a closer look. I guess the FPN weights  is not tracked by model correctly. Thanks for the report.

> 3. `mixed precision` makes loss value to `nan` (probably need to cast the output of segmentation head to `tf.float32`. You might want to introduce `dtype` param in [here](https://github.com/keras-team/keras-cv/blob/673e63cf2fc5b4ff295004a20aac7b5e2544c16d/keras_cv/models/segmentation/__internal__/segmentation_head.py#L51).)

Thanks. Will do.

> 
> [Reproducible Code.](https://colab.research.google.com/drive/1x-mB_CPnQkuqSMq7peHWk32h6V9Qi6h3?usp=sharing)

**Comment by qlzh727** [comment]
The PR should address 1 and 3. 

For 2, after a closer look, this warning is intended. For the FPN, only the output at level 2 is consumed by the segmentation head, which is why the P3~P5 weights doesn't have any loss. Since the FPN is decoupled from the head, it doesn't know that only part of its output are consumed.

**Comment by innat** [comment]
@qlzh727 
Thanks for addressing these issues. I did a quick test and the main issues are gone. I compared the results of first run with similar [codebase](https://www.tensorflow.org/tutorials/images/segmentation), and segmentation resutls are achieved as follows. The results are pixeled but it might need better training care. I will check later.

![download](https://user-images.githubusercontent.com/17668390/193917368-11ba6fb3-b81a-4661-8813-bd8bef87cc79.png)
![download](https://user-images.githubusercontent.com/17668390/193917329-c42b2585-66fe-4fec-8a0c-3d52079b4908.png)


About warning message, 

> For 2, after a closer look, this warning is intended. For the FPN, only the output at level 2 is consumed by the segmentation head, which is why the P3~P5 weights doesn't have any loss. Since the FPN is decoupled from the head, it doesn't know that only part of its output are consumed.

I think it would be nice to hind this warning message. It's not only unnecessary warning but also confusing for end users (IMO).  this should be also documented clearly.
